### ISSUE TITLE
No way in 6.x to build our own JSON and use LowLevel client to Index

### ISSUE THREAD

**Issue by onspring-technologies** [elastic/elasticsearch-net#3281]
**NEST/Elasticsearch.Net version**: 6.1

**Elasticsearch version**: 6.x

**Description of the problem including expected versus actual behavior**:

**Steps to reproduce**:
 1. New Nuget packages installed & New elastic installed clean

**Provide `ConnectionSettings` (if relevant)**:

**Provide `DebugInformation` (if relevant)**:

**Describe the feature**:

In the old version (5), we were able to build our own JSON at about 10x the speed of serialization & deserialization. For this reason, we were using the LowLevel features to do something to the effect of this:

            var doc = BuildJsonDoc(record);
            Client.LowLevel.Index<dynamic>(IndexName, TypeName, record.Id.ToString(), doc);

Our "Record" type is extremely complex and does not belong, in whole, within elastic. As such, building the Json from the relevant pieces and parts is fast. The structure of our Record type is also significantly dynamic, in that we do not know 90% of the fields ahead of time. We parse them out of an array of other objects. 

For elastic performance sake, simply importing a similar array structure is horrible. Thus, our method of stripping down our "Record" saves both 10x on import and about 6x on search performance.

Basically, we'd just like to have the version 6.x of Client.LowLevel, that removes the "Type" requirement (not used in 6.x), and allows the doc indexing to work properly.

**Comment by russcam** [comment]
@onspring-technologies is the issue slower serialization/deserialization is 6.x compared to 5.x? Can you share the connection settings  and client configuration you're using?

**Comment by onspring-technologies** [comment]
No, the issue is that the absolute fastest way to create the object we want to store in elastic is to simply build the Json with a Json builder. Creating a dynamic object and serializing it is much slower.

**Comment by russcam** [comment]
So I understand correctly, a custom type you have i.e. `Record` serializes/deserializes faster than `dynamic`?

**Comment by onspring-technologies** [comment]
No. We are building json from scratch, because our document is insanely simple, and much much faster serializing anything, especially the Record object, which has too much info. So, we simplify and only pull relevant, searchable data.  We build json from scratch, because it's faster than making that simple document as a dynamic object first, then serializing

**Comment by russcam** [comment]
> We build json from scratch, because it's faster than making that simple document as a dynamic object first, then serializing

Serializing a JSON string or byte array of JSON would be much faster than serializing a `dynamic` type with JSON.Net, which would need to reflect over properties. I'm not sure what Json.NET does in the way of caching for `dynamic` types; it could be that no contracts for the dynamic structure are cached after first invocation and therefore reflection is happening on each call.

Do you have a small but complete reproducible example that demonstrates what you're seeing, and what you're doing instead?

**Comment by russcam** [comment]
I'm just trying to fully understand what the issue is. It seemed based on the original description that the issue was related to serialization performance, but you're saying that's not the case? 

> We just want to keep using the insert function on the low level client but sending it the string json document with the index name and Id.

The `type` parameter won't be going away in NEST/Elasticsearch.Net 6.x, since removing it now would break backwards binary compatibility for the client in a major version. If you don't care what the value for type is, you can use `"doc"` in Elasticsearch 6.0.x and 6.1.x or `"_doc"` in Elasticsearch 6.2.x + for the document type

**Comment by russcam** [comment]
You can send a raw JSON string in pretty much the same way as in 5.x
```c#
var client = new ElasticClient();    
var json = "{ \"field\": \"value\" }";    
client.LowLevel.Index<BytesResponse>("index", "type", json);
```
`string` implicitly converts to `PostData` and serializes with

https://github.com/elastic/elasticsearch-net/blob/96dff79b75eecf46cb5db927060166bda979edd7/src/Elasticsearch.Net/Transport/PostData.cs#L93-L94

I think we can forgo writing to a `MemoryStream` then copying to the request stream though, when `DisableDirectStreaming` is `false` (the default).

**Comment by onspring-technologies** [comment]
I'm getting errors that won't allow compilation because it says it can't implicitly convert

**Comment by russcam** [comment]
Would you be able to provide a small but complete reproducible example? Ensure that both Elasticsearch.Net and NEST are both version 6.1.0

**Comment by onspring-technologies** [comment]
var doc = GetIndexRecordJson(app, record); // doc is a stringbuilder
            Client.LowLevel.Index<dynamic>(IndexName, app.Id.ToString(), record.Id.ToString(), doc);

We're getting "cannot convert from string to" errors. If we try to solve by adding a "null" in after the doc, we get an "implicit reference conversion" issue.

**Comment by Mpdreamz** [comment]
Don't use `dynamic` as response type use `DynamicResponse`. 

See https://www.elastic.co/guide/en/elasticsearch/client/net-api/master/elasticsearch-net-breaking-changes.html#_response_types

**Comment by onspring-technologies** [comment]
Tested and verified that DynamicReponse worked.

**Comment by russcam** [comment]
@onspring-technologies Great! Are you happy for us to close this issue?

**Comment by onspring-technologies** [comment]
Go for it!
